src/components/switch.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import { css, Input } from 'theme-ui';
+import { applyVariation } from './../utils/getStyles';
+
+const defaultVariants = {
+  sm: {
+    width: '30px',
+    height: '15px',
+    borderColor: 'borderGray',
+    bg: 'borderGray',
+    color: 'white',
+  },
+  md: {
+    width: '38px',
+    height: '19px',
+    border: 'borderGray',
+    bg: 'borderGray',
+    color: 'white',
+  },
+  lg: {
+    width: '44px',
+    height: '22px',
+    border: 'borderGray',
+    bg: 'borderGray',
+    color: 'white',
+  },
+};
+
+const getVariants = (theme, variant = 'md', key) =>
+  theme.switch[variant] && theme.switch[variant][key]
+    ? theme.switch[variant][key]
+    : defaultVariants[variant][key];
+
+const Checkbox = styled(Input)`
+  ${({ theme, variant }) =>
+    css({
+      position: 'relative',
+      width: getVariants(theme, variant, 'width'),
+      height: getVariants(theme, variant, 'height'),
+      bg: getVariants(theme, variant, 'bg'),
+      WebkitAppearance: 'none',
+      outline: 'none',
+      borderRadius: '50px',
+      transition: '.5s',
+      borderWidth: '1px',
+      borderStyle: 'solid',
+      borderColor: 'transparent',
+      boxShadow: 'inset 0 0 5px rgba(0, 0, 0, .3)',
+      '&:checked': {
+        bg: 'primary',
+      },
+      '&::before': {
+        content: '""',
+        position: 'absolute',
+        borderRadius: '50px',
```

Comment:
should not it be referred from 'radii' from theme?

Reply:
No, if our theme is [4,6,8,50], referred to 3 and the user theme is some other it will not look like our default switch. if the user want to override he/she can from the theme.